cmd/jujud/agent/model.go

```diff
@@ -26,22 +52,103 @@ func (m *ModelCommand) Info() *cmd.Info {
 }
 
 // Init initializers the command for running
-func (m *ModelCommand) Init(_ []string) error {
+func (m *ModelCommand) Init(args []string) error {
+	if m.ModelUUID == "" {
+		return cmdutil.RequiredError("model-uuid")
+	}
+
+	if err := m.AgentConf.CheckArgs(args); err != nil {
+		return err
+	}
+
+	m.runner = worker.NewRunner(worker.RunnerParams{
+		IsFatal:       cmdutil.IsFatal,
+		MoreImportant: cmdutil.MoreImportant,
+		RestartDelay:  jworker.RestartDelay,
+	})
 	return nil
 }
 
+// maybeCopyAgentConfig copies the read-only agent config template
+// to the writeable agent config file if the file doesn't yet exist.
+func (m *ModelCommand) maybeCopyAgentConfig() error {
```

Comment:
this would be common with the app operator but maybe not worth exporting and sharing for reuse

Reply:
My understanding is that this is needed inside the docker image.

Reply:
@wallyworld ? Does this need to be changed based on above?